urllib3/connectionpool.py

```diff
@@ -666,6 +666,10 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                                 release_conn=release_conn, body_pos=body_pos,
                                 **response_kw)
 
+        def drain_and_release_conn(response):
+            response.read()
```

Comment:
The general concern is that we have, in the past, bumped into malformed responses in the past, including on redirects.